pdr_backend/subgraph/subgraph_predictions.py

```diff
@@ -112,12 +113,19 @@ def fetch_filtered_predictions(
                 }}
             }}"""
 
-        print("Querying subgraph...", query)
-        result = query_subgraph(
-            get_subgraph_url(network),
-            query,
-            timeout=20.0,
-        )
+        try:
+            print("Querying subgraph...", query)
+            result = query_subgraph(
+                get_subgraph_url(network),
+                query,
+                timeout=20.0,
+            )
+        except Exception as e:
```

Comment:
```
Imo, it's better to have no data at all than to work with partial or incomplete data.
```
Definitely not how I've done things in the past...

Assume the system has made 100s of successful fetches, why not save that, and then have the system resume from the last successful record/fetch?

I'd rather assume a dumb system that saves whatever it can fetch successfully and resumes from the last_save, than to custom-tweak each subgraph for a max_fetch/parameters such that it "has to succeed, to save".

```
As mentioned in my previous comment, the current handling of this error is inadequate.
```
Are there specific errors (i.e 502) that you think should be uniquely handled, or is this more of a comment as a whole towards "don't save if there is an error".

```
Ultimately, however, the final decision rests with you sir!
```
I think this implementation is in-line with what me and @trentmc have discussed in the past.

Reply:
Spoke w/ trizin, moving any remaining discussions to slack, and merging the PR.